src/core.ts

```diff
@@ -355,36 +355,68 @@ export type TrackingConsent = {
   [purpose: string]: boolean | undefined;
 };
 
-/** Tracking purpose consent config with timestamp & auth metadata */
-export type TrackingConsentDetails = {
-  /** Tracking consent config */
-  purposes: TrackingConsent;
-  /**
-   * Was tracking consent confirmed by the user?
-   * If this is false, the consent was resolved from defaults & is not yet confirmed
-   */
-  confirmed: boolean;
-  /** Consent resolution/last-modified timestamp (ISO 8601) */
-  timestamp: string;
-  /** Has the consent been updated (including no-change confirmation) since default resolution */
-  updated?: boolean;
-  /** Whether or not the UI has been shown to the end-user (undefined in older versions of airgap.js) */
-  prompted?: boolean;
-  /** Transparency Consent (TCF) String */
-  tcf?: string;
-  /** US Privacy (USP) String */
-  usp?: string;
-  /** Global Privacy Platform (GPP) String */
-  gpp?: string;
-  /** Consent Manager View State */
-  viewState?: ViewState;
-  /** Airgap Version */
-  airgapVersion?: string;
-  /** Arbitrary metadata that customers want to be associated with consent state */
-  metadata?: unknown;
-  /** When the metadata was last updated */
-  metadataTimestamp?: string;
-};
+export const TrackingConsentDetails = t.intersection([
+  t.type({
+    /** Tracking consent config */
+    purposes: t.record(t.string, t.boolean),
+    /**
+     * Was tracking consent confirmed by the user?
+     * If this is false, the consent was resolved from defaults & is not yet confirmed
+     */
+    confirmed: t.boolean,
+    /** Consent resolution/last-modified timestamp (ISO 8601) */
+    timestamp: t.string,
+  }),
+  t.partial({
+    /** Has the consent been updated (including no-change confirmation) since default resolution */
+    updated: t.boolean,
+    /** Whether or not the UI has been shown to the end-user (undefined in older versions of airgap.js) */
+    prompted: t.boolean,
+    /** Transparency Consent (TCF) String */
+    tcf: t.string,
+    /** US Privacy (USP) String */
+    usp: t.string,
+    /** Global Privacy Platform (GPP) String */
+    gpp: t.string,
+    /** Consent Manager View State */
+    viewState: valuesOf(ViewState),
+    /** Airgap Version */
+    airgapVersion: t.string,
+    /** Arbitrary metadata that customers want to be associated with consent state */
+    metadata: t.unknown,
+    /** When the metadata was last updated */
+    metadataTimestamp: t.string,
+  }),
+]);
+
+/** Override types. */
+export type TrackingConsentDetails = t.TypeOf<typeof TrackingConsentDetails>;
+
+export const ConsentPreferencesBody = t.type({
+  /** token containing encrypted identifier */
+  token: t.string,
+  /** parent organization  */
+  organizationId: t.string,
```

Comment:
@anotherminh , chatted with @eligrey and I think if the partition ID is going to include the parent org ID—particularly if we are not enforcing universal uniqueness on the partition key itself—we should include the parent org ID in the `ConsentPreferencesBody`. Does that make sense to you? (We'll also have to include this in the bundle as well, of course.)